### ISSUE TITLE
Revamp GitHub labels

### ISSUE THREAD

**Issue by akihironitta** [PyTorchLightning/pytorch-lightning#10473]
## 🚀 Feature

- [x] 1. Add/edit/remove labels
- [ ] 2. Check/update configs which uses outdated labels
  - https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/mergify.yml
  - https://github.com/PyTorchLightning/pytorch-lightning/tree/master/.github/ISSUE_TEMPLATE
  - https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/stale.yml

### Motivation
To address the linked issues.

### Pitch

https://docs.google.com/document/d/1JeCC0PadWVD84wYCBMRyParDpdz3oMfhJwQFrp1CD0c/edit?usp=sharing

### Additional context
This issue should be addressed before:
- #9519
- #9216

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

**Comment by akihironitta** [comment]
Could someone update labels at https://github.com/PyTorchLightning/pytorch-lightning/labels?

**Comment by kaushikb11** [comment]
> Pitch
According to the Google doc.

@akihironitta Where's the doc?

> Could someone update labels at PyTorchLightning/pytorch-lightning/labels?

I could. What's the goal?

**Comment by carmocca** [comment]
Let me take care of it as I wrote the document :P

**Comment by carmocca** [comment]
The labels have been updated. @akihironitta do you want to open the PR updating the configs?
